extensions/notion/src/components/actions/ActionCreateQuicklink.tsx

```diff
@@ -0,0 +1,24 @@
+import { Action } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+
+import { getPageName } from "../../utils/notion";
+import { getOpenIn } from "../../utils/openPage";
+import { Page } from "../../utils/types";
+
+export default function ActionCreateQuicklink({ page }: { page: Page }) {
+  if (!page.url) return null;
+
+  const { data } = useCachedPromise(() => getOpenIn());
```

Comment:
you'll run the promise for every time this action is rendered. You probably should just compute it once and cache it in memory (no need to cache it on a file)